web/src/project-activity.js

```diff
@@ -171,96 +168,109 @@ class ProjectActivity extends Component {
+          <br />
+          <br />
           <table>
             <tbody>
-              {successArr.map(e => (<tr><td>{e.name}</td></tr>))}
+              {successArr.map(e => (
+                <tr>
+                  <td>{e.name}</td>
+                </tr>
+              ))}
             </tbody>
           </table>
           <br />
-          <span className='supporting'>{SYSTEM_RESTART_RECOMMENDATION}</span>
-          <br /><br />
+          <span className="supporting">{SYSTEM_RESTART_RECOMMENDATION}</span>
+          <br />
+          <br />
         </div>
       );
     } else {
       success = (
         <div>
-          <span className='project-activity-update-modal-section-header'>No updates</span>
-          <br /><br />
-          <span className='project-activity-update-modal-section-message'>
+          <span className="project-activity-update-modal-section-header">No updates</span>
+          <br />
+          <br />
+          <span className="project-activity-update-modal-section-message">
             The latest versions are already installed
           </span>
-          <br /><br />
+          <br />
+          <br />
         </div>
       );
     }
 
-    let failure = undefined;
+    let failure;
     if (failureArr && failureArr.length) {
       failure = (
         <div>
-          <span className='project-activity-update-modal-section-header'>Failed</span>
-          <br /><br />
+          <span className="project-activity-update-modal-section-header">Failed</span>
+          <br />
+          <br />
           <table>
             <tbody>
               {failureArr.map(e => (
                 <tr>
                   <td style={{ width: '100px' }}>{e.name}</td>
                   <td style={{ width: 'auto' }}>{e.failureResult.error}</td>
-                </tr>))}
+                </tr>
+              ))}
             </tbody>
           </table>
         </div>
       );
     }
 
     return (
-      <ModalContent
-        buttonAction={this.modalDismiss}
-        confirmOnly={true}
-      >
+      <ModalContent buttonAction={this.modalDismiss} confirmOnly>
         {success}
         {failure}
       </ModalContent>
-    )
+    );
   };
 
-  handleUpdateAllAction = (projectList) => {
+  handleUpdateAllAction = projectList => {
     const modalCompletion = choice => {
       if (choice === 'ok') {
-        this.props.updateAllProjects(projectList,
+        this.props.updateAllProjects(
+          projectList,
           // completion callback
           (successArr, failureArr) => {
+            let success = successArr;
+            let failure = failureArr;
             if (successArr) {
-              successArr = successArr.sort(orderResultsByProjectName).filter(response => response.successResult.updated);
+              success = [...successArr]
+                .sort(orderResultsByProjectName)
+                .filter(response => response.successResult.updated);
```

Comment:
Fix an error caused by reassigning a function argument. I also created a shallow copy of the original argument because `sort` mutates the array in place.